controlplane/kubeadm/controllers/helpers.go

```diff
@@ -241,8 +243,34 @@ func (r *KubeadmControlPlaneReconciler) generateMachine(ctx context.Context, kcp
 		},
```

Comment:
Let's also remove the hash label

Reply:
Some tests are failing, could not figure out. Leaving this for a follow up PR.